ddl/ddl_api.go

```diff
@@ -305,20 +306,41 @@ func checkColumnDefaultValue(ctx sessionctx.Context, col *table.Column, value in
 	if value != nil && ctx.GetSessionVars().SQLMode.HasNoZeroDateMode() &&
 		ctx.GetSessionVars().SQLMode.HasStrictMode() && types.IsTypeTime(col.Tp) {
 		if vv, ok := value.(string); ok {
-			t, err := types.ParseTime(nil, vv, col.Tp, 6)
+			timeValue, err := expression.GetTimeValue(ctx, vv, col.Tp, col.Decimal)
```

Comment:
Why not col.Decimal? 6 is a little strange.

Reply:
I'm not sure whether this is correct.
You'd better check whether this will cause error, we can define a column like `col time(3) '0000-00-00 00:00:00.123456'`

Reply:
`col time(3) '0000-00-00 00:00:00.123456'` will return error in `getDefaultValue`. Won't fetch here.